documentation/asciidoc/accessories/camera/libcamera_apps_post_processing.adoc

```diff
@@ -0,0 +1,216 @@
+
+`libcamera-hello --post-process-file /path/to/negate.json`
+
+The negate stage is particularly trivial and has no configuration parameters of its own, therefore the JSON file merely has to name the stage, with no further information, and it will be run. Thus `negate.json` contains
+
+----
+{
+    "negate":
+    {
+    }
+}
+----
+
+To run multiple post-processing stages, the contents of the example JSON files merely need to be listed together, and the stages will be run in the order given. For example, to run the Sobel stage (which applies a Sobel filter to an image) followed by the negate stage we could create a custom JSON file containing
+
+----
+{
+    "sobel_cv":
+    {
+        "ksize": 5
+    },
+    "negate":
+    {
+    }
+}
+----
+
+The Sobel stage is implemented using OpenCV, hence `cv` in its name. Observe how it has a user-configurable parameter, `ksize` that specifies the kernel size of the filter to be used. In this case, the Sobel filter will produce bright edges on a black background, and the negate stage will turn this into dark edges on a white background, as shown.
+
+image::images/sobel_negate.jpg[Image with Sobel and negate]
+
+Some stages actually alter the image in some way, and this is their primary function (such as _negate_). Others are primarily for image analysis, and while they may indicate something on the image, all they really do is generate useful information. For this reason we also have a very flexible form of _metadata_ that can be populated by the post-processing stages, and this will get passed all the way through to the application itself.
+
+Image analysis stages often prefer to work on reduced resolution images. `libcamera-apps` are able to supply applications with a ready-made low resolution image provided directly by the ISP hardware, and this can be helpful in improving performance.
+
+Furthermore, with the post-processing framework being completely open, Raspberry Pi welcomes the contribution of new and interesting stages from the community and would be happy to host them in our `libcamera-apps` repository. The stages that are currently available are documented below.
+
+NOTE: the `libcamera-apps` supplied with the operating system will be built without any optional 3rd party libraries (such as OpenCV), meaning that certain post-processing stages that rely on them will not be enabled. To use these stages, please follow the instructions for xref:camera.adoc#building-libcamera-and-libcamera-apps[building `libcamera-apps` for yourself].
+
+==== `negate` stage
+
+The `negate` stage requires no 3rd party libraries.
+
+It may be more performant on Pi 3 and 32-bit OS Pi 4 devices if recompiled using `-DENABLE_COMPILE_FLAGS_FOR_TARGET=armv8-neon`. (Please see the xref:camera.adoc#building-libcamera-and-libcamera-apps[build instructions].)
```

Comment:
"32-bit OS Pi 4 devices" sounds a bit weird IMHO.
@aallan can confirm, but I think the recommended phrasing would be something like "On Raspberry Pi 3 and Raspberry Pi 4 computers running a 32-bit OS, it may be more performant if ..." ?

Reply:
Yup. We need to avoid using "Pi" anywhere in the documentation — in **all** cases it should be "Raspberry Pi".